src/Widgets/NotificationEntry.vala

```diff
@@ -61,6 +57,41 @@ public class Notifications.NotificationEntry : Gtk.ListBoxRow {
             app_image.icon_name = notification.app_icon;
         }
 
+        var image_overlay = new Gtk.Overlay ();
+        image_overlay.valign = Gtk.Align.START;
+
+        if (notification.image_path != null && notification.image_path != "") {
+            try {
+                var scale = get_style_context ().get_scale ();
+                var pixbuf = new Gdk.Pixbuf.from_file_at_size (notification.image_path, 48 * scale, 48 * scale);
```

Comment:
I think it could be a good idea to move those `48`s to a constant and share it with `MaskedImage.ICON_SIZE` or pass the icon size to the `MaskedImage` constructor instead of using a hardcoded value.

```suggestion
                var pixbuf_size = ICON_SIZE * scale;
                var pixbuf = new Gdk.Pixbuf.from_file_at_size (notification.image_path, pixbuf_size, pixbuf_size);
```

Reply:
The MaskedImage has to be hardcoded because it matches the size of the overlayed mask, but I've added constants here in NotificationEntry